scripts/dts/edtlib.py

```diff
@@ -773,6 +773,19 @@ def _init_binding(self):
                     self.matching_compat = compat
                     self._binding = binding
                     return
+
+                # If matching for the specific on_bus value failed look for
+                # bindings which don't have an explicitly defined bus (i.e.
+                # can support any bus).
+                for (compat, on_bus) in self.edt._compat2binding.keys():
+                    if on_bus is not None:
+                        continue
+                    if compat in self.compats:
+                        binding = self.edt._compat2binding[compat, on_bus]
+                        self.binding_path = binding.path
+                        self.matching_compat = compat
+                        self._binding = binding
+                        return
```

Comment:
Why can't you just look up `self.edt._compat2binding[compat, None]`?

Reply:
@mbolivar-nordic I certainly can. It's just that my python-foo wasn't quite sharp enough so I didn't think about that option. It'd look like the following then, I suppose?
```
if (compat, None) in self.edt._compat2binding:
    binding = self.edt._compat2binding[compat, None]
    self.binding_path = binding.path
    self.matching_compat = compat
    self._binding = binding
    return
```

I take it that you're not in general opposed to this feature then?

Reply:
Actually, I guess I can't use None in the variable portion of the if-statement. So I should just directly look up from the dictionary and handle the case (exception?) that the key isn't there?

Reply:
>  It'd look like the following then, I suppose?

Seems right to me. A test case in testedtlib.py verifying this would also help.

> I take it that you're not in general opposed to this feature then?

Seems reasonable to me given the rest of the discussion here that it's a standalone generalization not directly related to PCIe.

Reply:
> Actually, I guess I can't use None in the variable portion of the if-statement. 

Hmm, I just attempted a build with that and didn't get any errors, so maybe that is correct syntax after all (i.e. None means that the portion of the key gets ignored and not assigned to any variable)

Reply:
I don't have any issues with this:

```python
d = {('vnd,foo', 'i2c'): 'binding1', ('vnd,foo', None): 'binding2'}
('vnd,foo', None) in d  # True
d['vnd,foo', None]  #  'binding2'
```

In general, `d[foo, bar]` is the same as `d[(foo, bar)]`, i.e. multiple arguments get wrapped up in a tuple as syntactic sugar, and the `_compat2binding` dict contains `(compatible, bus)` tuples.

Reply:
>  A test case in testedtlib.py

If you can add these, I think I'm ready to +1

Reply:
> If you can add these, I think I'm ready to +1

Test cases have been added.